### PR TITLE
Updates to behaviour and style based on UX specs

### PR DESCRIPTION
**Pull request by rox163**
#minor#

### CODE REVIEW

File: addon/components/frost-text.js [hunk 1]
```diff
@@ -1,8 +1,13 @@
 import _ from 'lodash'
 import Ember from 'ember'
+import layout from '../templates/components/frost-text'
 
-export default Ember.TextField.extend({
+export default Ember.Component.extend({
```
Comment (rox163) [inline addon/components/frost-text.js]: doing this to customize the layout and add the clear button

File: addon/components/frost-text.js [hunk 2]
```diff
@@ -12,5 +17,18 @@ export default Ember.TextField.extend({
     if (_.isFunction(onInput)) {
       onInput({id, value})
     }
-  })
+    this.set('showClear', true)
+  }),
+
+  onFocus: Ember.on('focusIn', function (e) {
```
Comment (rox163) [inline addon/components/frost-text.js]: selects all text on focus

File: addon/components/frost-text.js [hunk 3]
```diff
@@ -12,5 +17,18 @@ export default Ember.TextField.extend({
     if (_.isFunction(onInput)) {
       onInput({id, value})
     }
-  })
+    this.set('showClear', true)
```
Comment (sglanzer) [inline addon/components/frost-text.js]: We should only show the clear if the length of the input value is 1 or greater

File: addon/styles/addon.scss [hunk 4]
```diff
@@ -1,38 +1,63 @@
 @import 'node_modules/ember-frost-css-core/scss/frost-core';
 @import 'node_modules/ember-frost-theme/scss/frost-theme';
 
-$border: 1px solid $frost-color-lgrey-5;
+$border: 1px solid $frost-color-grey-6;
```
Comment (sglanzer) [inline addon/styles/addon.scss]: I'd like to see theme variables set up for these colors - even if it's just variables in this file for now

File: addon/templates/components/frost-text.hbs [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{{input type="text" autofocus=autofocus disabled=disabled value=value placeholder=placeholder readonly=readonly}}
```
Comment (sglanzer) [inline addon/templates/components/frost-text.hbs]: Please line-wrap the options - we'll add template linting to cover this at a later stage

{{input
  type='text'
  autofocus=autofocus
  ...

File: addon/styles/addon.scss [hunk 6]
```diff
@@ -1,38 +1,63 @@
...
+    &:hover:enabled:read-write{
+      border: $focus-border;
+    }
   }
 
-  &.error {
-    border: $error-border;
-    margin: $error-margin;
+  .close {
+    position: relative;
```
Comment (sglanzer) [inline addon/styles/addon.scss]: I'm a little concerned about this for two reasons:
- What happens to any content on the right side of this if the close appears/disappears?  
- What happens if the text flows under the X?

Can we position it relative inside the input instead?

File: addon/styles/addon.scss [hunk 7]
```diff
@@ -1,38 +1,65 @@
...
+$focus-border: 2px solid $frost-color-input-focus-border;
 $focus-margin: 0;
+$disabled-border: 1px solid $frost-color-input-disabled-border;
+$disabled-bg: $frost-color-input-disabled-bg;
 $font-size: $frost-font-s;
-$height: 30px;
+$height: 35px;
 $margin: 1px; // Note: margin + border width equals border width of focus state
-$padding: 0 6px;
+$padding: 0 19px 0 6px;
```
Comment (rox163) [inline addon/styles/addon.scss]: the 'padding-right' here ensures the text doesn't flow under the X.

File: package.json [hunk 8]
```diff
@@ -45,7 +45,8 @@
     "ember-export-application-global": "^1.0.5",
     "ember-frost-button": "^1.4.0",
     "ember-frost-css-core": "^1.1.2",
-    "ember-frost-theme": "^1.2.0",
+    "ember-frost-icons": "^1.4.0",
+    "ember-frost-theme": "^1.3.0",
```
Comment (sglanzer) [inline package.json]: Make sure you put these changes into the blueprint dependency list as well

**Comment by sglanzer** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-text/pull-request/12/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-text)

**Pull request merged**